### DISCUSSION TITLE
Declare without initializing

### DISCUSSION THREAD

**Discussion by freedxmgxd** [carbon-language/carbon-lang#2135] [General]
Will carbon let declare without initializing? It's seen to me as something that causes more trouble than is useful, for unexpected behavior. Since it's possible in Cpp, I have not seen good reasons to maintain this functionality.

**Comment by micttyl** [comment]
If I understand both Carbon and you correctly, Carbon leaves this up to the users, so the programmers decide it when they are coding.

**Comment by jonmeow** [comment]
I think proposal #257 still needs to be integrated into the design, but see [unformed objects](https://github.com/carbon-language/carbon-lang/blob/trunk/proposals/p0257.md#unformed-objects) in it.
